subprojects/packagefiles/nanobind/meson.build

```diff
@@ -0,0 +1,87 @@
+project(
+    'nanobind',
+    'cpp',
+    version: '2.0.0',
+    meson_version: '>=0.53.0',
+    license: 'BSD-3-Clause',
+)
+
+py_mod = import('python')
+py = py_mod.find_installation()
+py_dep = py.dependency(embed: true)
+
+robin_map_dep = dependency('robin-map')
+
+incdir = include_directories('include')
+
+# Arguments are derived from nanobind_build_library function in CMake configuration
+dep_compile_args = []
+dep_link_args = []
+if get_option('python.allow_limited_api')
+  py_version = py.language_version()
+
+  if py_version.version_compare('>=3.12')
+    add_project_arguments('-DPy_LIMITED_API=0x030C0000', language: 'cpp')
+  endif
+endif
+
+if get_option('buildtype') == 'shared'
+  # nanobind_link_options uses linker response files for either PyPy or
+  # CPython at this step - how do we support in Meson?
+  add_project_arguments('-DNB_BUILD', language: 'cpp')
+  dep_compile_args += ['-DNB_SHARED']
+
+  if host_machine.system() == 'darwin'
+    add_project_link_arguments('-Wl,-dead_strip', '-Wl,x', '-Wl,-S', language: 'cpp')
+  elif host_machine.system() != 'windows'
+    add_project_link_arguments('-Wl,-s', language: 'cpp')
```

Comment:
Shouldn't these also be part of dep_link_args?

Reply:
The current CMake configuration sets these as PRIVATE on the target, so just copying that over. Happy to move this to the dependency interface if you think that makes more sense

Reply:
Ah nevermind - I take that back. I see it is called by both the internal function to build the nanobind library as well as the `nanobind_add_module` function that end users are expected to use.

I think what you are proposing makes sense